test/hotspot/jtreg/compiler/vectorization/TestFloat16VectorOperations.java

```diff
@@ -0,0 +1,288 @@
+            }
+        }
+    }
+
+    @Test
+    @Warmup(10000)
+    @IR(counts = {IRNode.MIN_VHF, ">= 1"},
+        applyIfCPUFeature = {"avx512_fp16", "true"})
+    public void vectorMinFloat16() {
+        for (int i = 0; i < LEN; ++i) {
+            output[i] = float16ToRawShortBits(min(shortBitsToFloat16(input1[i]), shortBitsToFloat16(input2[i])));
+        }
+    }
+
+    @Check(test="vectorMinFloat16")
+    public void checkResultMin() {
+        for (int i = 0; i < LEN; ++i) {
+            short expected = floatToFloat16(Math.min(float16ToFloat(input1[i]), float16ToFloat(input2[i])));
+            if (output[i] != expected) {
+                throw new RuntimeException("Invalid result: output[" + i + "] = " + output[i] + " != " + expected);
+            }
+        }
+    }
+
+    @Test
+    @Warmup(10000)
+    @IR(counts = {IRNode.MAX_VHF, ">= 1"},
+        applyIfCPUFeature = {"avx512_fp16", "true"})
+    public void vectorMaxFloat16() {
+        for (int i = 0; i < LEN; ++i) {
+            output[i] = float16ToRawShortBits(max(shortBitsToFloat16(input1[i]), shortBitsToFloat16(input2[i])));
+        }
+    }
+
+    @Check(test="vectorMaxFloat16")
+    public void checkResultMax() {
+        for (int i = 0; i < LEN; ++i) {
+            short expected = floatToFloat16(Math.max(float16ToFloat(input1[i]), float16ToFloat(input2[i])));
+            if (output[i] != expected) {
+                throw new RuntimeException("Invalid result: output[" + i + "] = " + output[i] + " != " + expected);
+            }
+        }
+    }
+
+    @Test
+    @Warmup(10000)
+    @IR(counts = {IRNode.SQRT_VHF, ">= 1"},
+        applyIfCPUFeature = {"avx512_fp16", "true"})
+    public void vectorSqrtFloat16() {
+        for (int i = 0; i < LEN; ++i) {
+            output[i] = float16ToRawShortBits(sqrt(shortBitsToFloat16(input1[i])));
+        }
+    }
+
+    @Check(test="vectorSqrtFloat16")
+    public void checkResultSqrt() {
+        for (int i = 0; i < LEN; ++i) {
+            short expected = float16ToRawShortBits(valueOf(Math.sqrt(float16ToFloat(input1[i]))));
+            if (output[i] != expected) {
+                throw new RuntimeException("Invalid result: output[" + i + "] = " + output[i] + " != " + expected);
+            }
+        }
+    }
+
+    @Test
+    @Warmup(10000)
+    @IR(counts = {IRNode.FMA_VHF, ">= 1"},
+        applyIfCPUFeature = {"avx512_fp16", "true"})
+    public void vectorFmaFloat16() {
+        for (int i = 0; i < LEN; ++i) {
+            output[i] = float16ToRawShortBits(fma(shortBitsToFloat16(input1[i]), shortBitsToFloat16(input2[i]), shortBitsToFloat16(input3[i])));
+        }
+    }
+
+    @Check(test="vectorFmaFloat16")
+    public void checkResultFma() {
+        for (int i = 0; i < LEN; ++i) {
+            short expected = floatToFloat16(Math.fma(float16ToFloat(input1[i]), float16ToFloat(input2[i]), float16ToFloat(input3[i])));
```

Comment:
The expected for fma should be either implemented on similar lines as Float16.fma() or we could call Float16.fma here directly.